### PR TITLE
remove dbplyr conflicts

### LINKED ISSUE #177: Overlapping methods in connectapi break dbplyr functionality
Something unexpected is happening with connectapi's custom `tbl_lazy` methods for `op_vars`, `dim` and `dimnames` generics (some of which are probably not necessary with the latest version of dbplyr).

I have some code that relies on `colnames()`, where it's expected that `colnames()` returns the column names of the final table (if we take out `x`, then it shouldn't include `x`).

``` r
library(dplyr)
library(dbplyr)

mf <- memdb_frame(g = c(1:3), x = 3:1, y = 3:1 + 3)
mf
#> # Source:   table<dbplyr_001> [?? x 3]
#> # Database: sqlite 3.37.0 [:memory:]
#>       g     x     y
#>   <int> <int> <dbl>
#> 1     1     3     6
#> 2     2     2     5
#> 3     3     1     4
mf %>% colnames()
#> [1] "g" "x" "y"
mf %>% select(-x) %>% colnames()
#> [1] "g" "y"
```

But when I load `connectapi`, `colnames()` grabs the colnames from the wrong place (from the `x` of the select operation object), and gives an incorrect answer.

```r
library(connectapi)
#> Registered S3 methods overwritten by 'connectapi':
#>   method            from  
#>   dim.tbl_lazy      dbplyr
#>   dimnames.tbl_lazy dbplyr
mf %>% colnames()
#> [1] "g" "x" "y"
mf %>% select(-x) %>% colnames()
#> [1] "g" "x" "y"
```

Similarly, joining is weird:

``` r
library(dplyr)
library(dbplyr)

mf1 <- memdb_frame(g = c(1:3), x = 3:1)
mf2 <- memdb_frame(g = c(1:3), y = 3:1 + 3)

left_join(mf1, mf2)
#> Joining, by = "g"
#> # Source:   lazy query [?? x 3]
#> # Database: sqlite 3.37.0 [:memory:]
#>       g     x     y
#>   <int> <int> <dbl>
#> 1     1     3     6
#> 2     2     2     5
#> 3     3     1     4

library(connectapi)
#> Registered S3 methods overwritten by 'connectapi':
#>   method            from  
#>   dim.tbl_lazy      dbplyr
#>   dimnames.tbl_lazy dbplyr
left_join(mf1, mf2)
#> Joining, by = "g"
#> Error in UseMethod("op_vars"): no applicable method for 'op_vars' applied to an object of class "c('op_join', 'op_double', 'op')"
```

I think both cases trace back to the `op_vars` methods in connectapi, and from [looking at dplyr's source](https://github.com/tidyverse/dbplyr/blob/70f408517a20514aa68c87d356e2c7f272d8f562/R/tbl-lazy.R#L108), I'm not sure if the overlapping methods are even necessary anymore.

### PR DESCRIPTION
**Pull request by fh-mthomson**
Closes #177 

I don't have the ability to test this, but submitting a PR to start the conversation in case a release is prioritized! 

FYI @atheriel this conflict with `dbplyr` has been a subtle but nagging blocker for us trying to consume data from Snowflake when deploying to Connect. 

Given this, I've opted to avoid using `connectapi`, if possible, when running analyses that rely on `dbplyr`

### CODE REVIEW

File: R/lazy.R [hunk 1]
```diff
@@ -116,7 +116,7 @@ print.tbl_connect <- function(x, ..., n = NULL) {
 
 #' @export
 as.data.frame.tbl_connect <- function(x, row.names = NULL, optional = NULL, ..., n = Inf) {
-  as.data.frame(collect(x, n = n))
+  as.data.frame(dplyr::collect(x, n = n))
 }
 
 op_base_connect <- function(x, vars) {
```
Comment (fh-mthomson) [inline R/lazy.R:122]: I'm not familiar enough w/ this space, so started w/ only the directly-conflicting methods! :)

File: R/lazy.R [hunk 2]
```diff
@@ -157,14 +157,3 @@ op_vars.op_single <- function(op) op_vars(op$x)
 #' @export
 op_vars.tbl_lazy <- function(op) op_vars(op$ops)
 
-# important for `nrow`/`ncol` to work
```
Comment (fh-mthomson) [inline R/lazy.R:160]: deleting an "important"-commentd function probably merits further scrutiny!

**Comment by colearendt** [comment]
Thanks so much for the work here! And apologies for the issues that you ran into with these conflicts! The fix in #210 should resolve these 😄

**Comment by fh-mthomson** [comment]
Thank you!!

**Pull request closed without merging**